File: mobileBE/src/main/java/com/uth/mobileBE/dto/response/ViolationResponse.java
package com.uth.mobileBE.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.uth.mobileBE.models.enums.StatusViolation;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ViolationResponse {
    private Long violationId;
    private Long readerId;
    private String readerName;
    private String barcode;

    private String reason;
    private String status; // Nên dùng Enum thay vì String

    // Thêm thông tin sách để người dùng biết vi phạm liên quan đến sách nào
    private Long bookId;
    private String bookTitle;

    // Thêm thông tin mượn sách
    private Long loanId;
    private Long loanDetailId;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;

}

File: app/src/main/java/com/example/quanlythuvien/data/model/response/ViolationResponse.kt
package com.example.quanlythuvien.data.model.response


import com.google.gson.annotations.SerializedName

data class ViolationResponse(
    @SerializedName("violationId")
    val violationId: Long,

    @SerializedName("readerId")
    val readerId: Long,

    @SerializedName("readerName")
    val readerName: String,

    @SerializedName("reason")
    val reason: String,

    @SerializedName("status")
    val status: String,

    @SerializedName("bookTitle")
    val bookTitle: String?,

    @SerializedName("createdAt")
    val createdAt: String
)